autogen/variables.tf

```diff
@@ -300,6 +300,34 @@ variable "pod_security_policy_config" {
     "enabled" = false
   }]
 }
+
+variable "authenticator_groups_config" {
```

Comment:
Instead of exposing this as a list, can we simply expose `authenticator_groups_security_group` directly?

Reply:
Then we need to fix psp as well.  It is the exact same way.  The challenge is that sometimes the data model gets changed, for example Isto.  We then have to implement that change in the upstream data model as well.  Our istio config values are already out of date in the beta module.

Let me know exactly what you want me to do, and I will focus on it tomorrow.

Reply:
psp can be fixed as a separate PR. Please go ahead and update this PR only for the variables you added.

Yes, the underlying data model sometimes changes but I'd rather have us take on the burden of updating with those changes than forcing every module user to update it themselves.

Reply:
I understand that,  I am not certain we can have an empty or default value for say `workload_identity_config`.

With pod security policies we have this:

https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/87bdc01f5325c14163cf0e5ba9e03d87857eafd3/modules/beta-public-cluster/variables.tf#L277-L283

Which I think we can set a default variable easily.

With  `workload_identity_config` we need to have something like this:

```
workload_identity_config = [ 
  {
     identity_namespace = "GKE_METADATA_SERVER"
  }
]
```

The values can be "UNSPECIFIED", "SECURE", "EXPOSE", "GKE_METADATA_SERVER".

We can default a value for them, but again we have to set something, because of the data model the upstream provider has chosen.

I would prefer a bit more complex data model on our end that would not require constant updates, and document it a ton.  I found the values in the provider and can map them to the `--workload-metadata-from-node` flag in gcloud.

I understand your concern, just trying to work on a good design that is lower maintenance, and still usable by our users.

Reply:
> I understand your concern, just trying to work on a good design that is lower maintenance, and still usable by our users.

Even if we document it, I would much prefer that we take on the maintenance burden rather than customers.

If the field format changes, then it's preferable for us to update it in the module (once) rather than forcing every module user to update (potentially hundreds of times).

Admittedly, there are some cases where setting an appropriate default value is challenging but in those I think we can still use a conditional to pass an empty list.

Please proceed with exposing the higher-level options to users (ex. `workload_identity_namespace`) and transforming them into the correct values within the provider itself.

Reply:
### Current Status

Because we cannot create default values for `authenticator_groups_config`, I am recommending to have the user set the list by hand. I have not been able to figure out how to use a list and a map with a trinary, without major code complexity.  There are solutions, but they are really, really, really ugly complex code.  With tf 0.12, it becomes a lot easier, but we are not there yet.

You get the following type of error:

```hcl
Error: module.gke.google_container_cluster.primary: 1 error(s) occurred:

* module.gke.google_container_cluster.primary: At column 3, line 1: conditional operator cannot be used with map values in:

${var.authenticator_groups_config != "" ? map("security_group", var.authenticator_groups_config) : map()}
```

If you want to read more https://github.com/hashicorp/terraform/issues/12453 - it is a feature request that they implemented in 0.12.

A single list is not bad: https://github.com/hashicorp/terraform/issues/12453#issuecomment-284273475 with 0.11

Just a map alone is complex https://github.com/hashicorp/terraform/issues/12453#issuecomment-327266951

And we have a list with a map inside of it 🤦‍♂.

### Recommendation

`workload_identity_namespace` and `vertical_pod_autoscaling` are very doable.  I recommend `default authenticator_groups_config` default to an empty list, and then the user has to set it to a 

```hcl
default authenticator_groups_config = [
  {
    security_group = "my-security-group-name"
  }
]
```

while `workload_identity_namespace` or `vertical_pod_autoscaling` would be a simpler value like:

```hcl
workload_identity_namespace = "GKE_METADATA_SERVER"
vertical_pod_autoscaling = true
```

Reply:
@chrislovecnm In the interest of moving forward, I'd suggest we initially merge only `workload_identity_namespace` and `vertical_pod_autoscaling` in this PR, then do a follow-up PR for the authenticator groups config. Does that seem acceptable?

Reply:
@morgante  - @aaron-lane mentioned that he would prefer the changes in the 0.12 branch.  Has that changed?

Reply:
I defer to @aaron-lane but our top priority is 0.12 support.

Reply:
#177 is on the cusp of being merged, and released tomorrow. :crossed_fingers: